File: common/src/main/java/org/tdf/common/types/Chained.java
package org.tdf.common.types;

import org.tdf.common.util.HexBytes;

public interface Chained extends Hashed {
    HexBytes getHashPrev();

    default boolean isParentOf(Chained another) {
        return another.isChildOf(this);
    }

    default boolean isChildOf(Chained another) {
        return isChildOf(another.getHash());
    }

    default boolean isChildOf(HexBytes hash) {
        return getHashPrev().equals(hash);
    }
}

File: common/src/main/java/org/tdf/common/types/Chained.kt
package org.tdf.common.types

import org.tdf.common.util.HexBytes

interface Chained : Hashed {
    val hashPrev: HexBytes
    fun isParentOf(another: Chained): Boolean {
        return another.isChildOf(this)
    }

    fun isChildOf(another: Chained): Boolean {
        return isChildOf(another.hash)
    }

    fun isChildOf(hash: HexBytes): Boolean {
        return hashPrev == hash
    }
}